pkg/cli/cmd/taskrun/logs.go

```diff
@@ -0,0 +1,155 @@
+package taskrun
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/tektoncd/results/pkg/cli/options"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/results/pkg/cli/client"
+	"github.com/tektoncd/results/pkg/cli/client/logs"
+	"github.com/tektoncd/results/pkg/cli/client/records"
+	"github.com/tektoncd/results/pkg/cli/common"
+	"github.com/tektoncd/results/pkg/cli/config"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+)
+
+// logsCommand returns a cobra.Command that logs a TaskRun.
+func logsCommand(p common.Params) *cobra.Command {
+	opts := &options.LogsOptions{
+		ResourceType: common.ResourceTypeTaskRun,
+	}
+
+	eg := `Get logs for a TaskRun named 'foo' in the current namespace:
+  tkn-results taskrun logs foo
+
+Get logs for a TaskRun in a specific namespace:
+  tkn-results taskrun logs foo -n my-namespace
+
+Get logs for a TaskRun by UID if there are multiple TaskRun with the same name:
+  tkn-results taskrun logs --uid 12345678-1234-1234-1234-1234567890ab
+
+Get logs for a TaskRun from all namespaces:
+  tkn-results taskrun logs foo -A
+`
+
+	cmd := &cobra.Command{
+		Use:   "logs [taskrun-name]",
+		Short: "Get logs for a TaskRun",
+		Long:  "Get logs for a TaskRun by name or UID. If --uid is provided, the TaskRun name is optional.",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		Args: func(_ *cobra.Command, args []string) error {
+			// If UID is provided, no arguments are required
+			if opts.UID != "" {
+				return nil
+			}
+			// Otherwise, require exactly one argument
+			if len(args) != 1 {
+				return fmt.Errorf("requires exactly one argument when --uid is not provided")
+			}
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			allNs, _ := cmd.Flags().GetBool("all-namespaces")
+			nsSet := cmd.Flags().Changed("namespace")
+			if allNs && nsSet {
+				return errors.New("cannot use --all-namespaces/-A and --namespace/-n together")
+			}
+			c, err := config.NewConfig(p)
+			if err != nil {
+				return err
+			}
+			opts.Client, err = client.NewRESTClient(c.Get())
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			if len(args) > 0 {
+				opts.ResourceName = args[0]
+			}
+
+			// Build filter string to find the TaskRun
+			filter := common.BuildFilterString(opts)
+
+			// Handle namespace
+			parent := fmt.Sprintf("%s/results/-", p.Namespace())
+			if opts.AllNamespaces {
+				parent = "*/results/-"
+			}
+
+			// Create record client
+			recordClient := records.NewClient(opts.Client)
+
+			// Find the TaskRun record
+			resp, err := recordClient.ListRecords(ctx, &pb.ListRecordsRequest{
+				Parent:   parent,
+				Filter:   filter,
+				PageSize: 10,
```

Comment:
Will that iterate on pages and find the record if it is not in the first 10 entries? If I read the code correctly, it won't and that's a problem.

Reply:
Yes, I think it will iterate the pages and find the record. Since we are already using the filter here which will have TaskName or UID. So, PageSize is for returning the output based on the filter passed. i.e it will return only those records which is related to the provided TaskName or UID.
and I think that's the rare case when the records list goes beyond 10.

Let me know if you think we should increase the page size limit.

Reply:
Yeah, I don't see a downside if I increase the limit.
I will do it 25 in that case. Thank you for pointing that out.